src/main/java/org/apache/cassandra/distributed/api/LogAction.java

```diff
@@ -0,0 +1,382 @@
+    {
+
+        return watchFor(startPosition, timeout, Internal.regexPredicate(pattern, others));
+    }
+
+    default LogResult<List<String>> watchFor(Duration timeout, String pattern, String... others) throws TimeoutException
+    {
+
+        return watchFor(Internal.DEFAULT_START_POSITION, timeout, Internal.regexPredicate(pattern, others));
+    }
+
+    default LogResult<List<String>> watchFor(long startPosition, String pattern, String... others) throws TimeoutException
+    {
+
+        return watchFor(startPosition, Internal.DEFAULT_TIMEOUT, Internal.regexPredicate(pattern, others));
+    }
+
+    default LogResult<List<String>> watchFor(String pattern, String... others) throws TimeoutException
+    {
+        return watchFor(Internal.DEFAULT_START_POSITION, Internal.DEFAULT_TIMEOUT, Internal.regexPredicate(pattern, others));
+    }
+
+    default LogResult<List<String>> grep(long startPosition, Predicate<String> fn)
+    {
+        try (LineIterator it = match(startPosition, fn))
+        {
+            return new BasicLogResult<>(it.mark(), Internal.collect(it));
+        }
+    }
+
+    default LogResult<List<String>> grep(Predicate<String> fn)
+    {
+        return grep(Internal.DEFAULT_START_POSITION, fn);
+    }
+
+    default LogResult<List<String>> grep(long startPosition, List<Pattern> patterns)
+    {
+        return grep(startPosition, Internal.regexPredicate(patterns));
+    }
+
+    default LogResult<List<String>> grep(List<Pattern> patterns)
+    {
+        return grep(Internal.DEFAULT_START_POSITION, Internal.regexPredicate(patterns));
+    }
+
+    default LogResult<List<String>> grep(long startPosition, Pattern pattern)
+    {
+        return grep(startPosition, Internal.regexPredicate(pattern));
+    }
+
+    default LogResult<List<String>> grep(Pattern pattern)
+    {
+        return grep(Internal.DEFAULT_START_POSITION, Internal.regexPredicate(pattern));
+    }
+
+    default LogResult<List<String>> grep(long startPosition, String pattern)
+    {
+        return grep(startPosition, Internal.regexPredicate(pattern));
+    }
+
+    default LogResult<List<String>> grep(String pattern)
+    {
+        return grep(Internal.DEFAULT_START_POSITION, Internal.regexPredicate(pattern));
+    }
+
+    default LogResult<List<String>> grep(long startPosition, String pattern, String... others)
+    {
+
+        return grep(startPosition, Internal.regexPredicate(pattern, others));
+    }
+
+    default LogResult<List<String>> grep(String pattern, String... others)
+    {
+        return grep(Internal.DEFAULT_START_POSITION, Internal.regexPredicate(pattern, others));
+    }
+
+    default LogResult<List<String>> grepForErrors(long startPosition)
+    {
+        Pattern logLevelPattern = Pattern.compile("^(INFO|DEBUG|WARN|ERROR)");
+        // WARN only included if this matches
+        Pattern exceptionPattern = Pattern.compile("[Ee]xception|AssertionError");
```

Comment:
Let's add an overload where this would be configurable?

Reply:
done.